File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] target = new int[26];
        String tmp = br.readLine();
        int len = tmp.length();
        for (int i = 0; i < len; i++) {
            target[tmp.charAt(i) - 65]++;
        }
        int ans = 0;

        outer:
        for (int i = 0; i < n - 1; i++) {
            String now = br.readLine();
            int nowLen = now.length();
            if (Math.abs(len - nowLen) > 1) {
                continue;
            }
            int[] arr = new int[26];
            for (int j = 0; j < nowLen; j++) {
                arr[now.charAt(j) - 65]++;
            }
            int diffCnt = 0;
            for (int j = 0; j < 26; j++) {
                if (arr[j] != target[j]) {
                    diffCnt++;
                }
            }
            if (len == nowLen) {
                if (diffCnt == 0) {
                    ans++;
                } else if (diffCnt == 2) {
                    for (int j = 0; j < 26; j++) {
                        if (arr[j] != target[j] && Math.abs(arr[j] - target[j]) > 1) {
                            continue outer;
                        }
                    }
                    ans++;
                }
            } else {
                if (diffCnt == 1) {
                    ans++;
                }
            }

        }
        System.out.println(ans);


        br.close();
    }
}

File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.py
import sys
sys.stdin.readline()
ans = 0
ref = [0] * 128
for i in sys.stdin.readline():
    ref[ord(i)] += 1
for l in sys.stdin:
    tmp = [0] * 128
    for i in l:
        tmp[ord(i)] += 1
    cnt1 = 0
    cnt2 = 0
    for i in range(65, 91):
        if ref[i] != tmp[i]:
            if ref[i] - tmp[i] == 1:
                cnt1 += 1
            elif ref[i] - tmp[i] == -1:
                cnt2 += 1
            else:
                cnt1 += 1000
                break
    if (cnt1 == 0 and cnt2 == 0) or(cnt1 == 1 and cnt2 == 1) or (cnt1 == 0 and cnt2 == 1) or (cnt1 == 1 and cnt2 == 0):
        ans += 1
print(ans)        